src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -278,6 +279,31 @@ public void SetAllExceptions(enum_EXCEPTION_STATE dwState)
             }
         }
 
+        public bool TryGetExceptionBreakpoint(string bkptno, out string exceptionName, out Guid exceptionCategoryGuid)
+        {
+            exceptionName = null;
+            exceptionCategoryGuid = Guid.Empty;
+            ExceptionCategorySettings categorySettings;
+            if (_categoryMap.TryGetValue(CppExceptionCategoryGuid, out categorySettings))
+            {
+                ulong breakpointNumber = Convert.ToUInt32(bkptno, CultureInfo.InvariantCulture);
+                lock (categorySettings.CurrentRules)
+                {
+                    if (categorySettings.CurrentRules.ContainsValue(breakpointNumber))
+                    {
+                        exceptionName = categorySettings.CurrentRules.First(pair => pair.Value == breakpointNumber).Key;
```

Comment:
nit: instead of traversing the collection twice, use:

```
string exceptionName = categorySettings.CurrentRules.FirstOrDefault(pair => pair.Value == breakpointNumber)?.Key;
if (exceptionName != null)
{
...
}
```